openid-4-verifiable-presentations-1_0.md

```diff
@@ -1464,6 +1464,180 @@ issuers in Self-Sovereign Identity ecosystems using TRAIN</title>
+
+## Protocol
+
+The value of the `protocol` parameter of the W3C Digital Credentials API MUST be set to `urn:openid.net:oid4vp' for this profile.
+
+## Request
+
+The `request` parameter of the W3C Digital Credentials API MUST contain a valid OID4VP Authorization Request, where every Authorization Request parameter is represented as a top-level JSON member as shown in the following non-normative example:
+
+```json
+{
+  "client_id": "client.example.org",
+  ...
+  "response_type": "vp_token",
+  "nonce": "n-0S6_WzA2Mj",
+  "presentation_definition": {...}
+}
+```
+
+The following Authorization Request parameters are supported with this profile: 
+
+* `client_id`
+* `client_id_scheme`
+* `response_type`
+* `nonce`
+* `presentation_definition`
+* `client_metadata`
+* `request`
+
+The `client_id` and `client_id_scheme` MUST be omitted in unsigned requests. The wallet determines the Client Identifier from the origin as asserted by the Browser. 
+
+This profile introduces a new parameter `expected_origins`.
+
+* `expected_origins`: An array of strings, each of the strings representing an origin of the Verifier making the request. This parameter MUST only be used with signed requests. It relates the logical Client Identifier to the physical endpoints that are legit origins for requests on behalf of this Client Identifier and is used to detect request replay.
+
+## Response
+
+Every OID4VP Authorization Request MUST result in a response being provided through the W3C Digital Credentials API. The Authorization Response is a JSON object, where the response parameters as defined for the Response Type are encoded as top-level claims in this JSON object. 
+
+The following is an example of an OID4VP Authorization Response through the API: 
+
+```json
+{
+  "presentation_submission": "...",
+  "vp_token": "..."
+}
+```
+
+Note: All mechanisms for cryptographically protecting the OID4VP response MAY be utilized with the W3C Digital Credentials API, too. 
+
+## Example Flows
+
+As OID4VP is pretty flexible, this section shall shed some light on what flows the authors deem especially useful.  
+
+### Unsigned Request
+
+The Verifier MAY send all the OID4VP request data as JSON elements in the `request` API parameter and reveices the result in the API's `result` parameter. In this case, the Wallet will use the Verifier origin as asserted by the Browser as the Verifers's Client Identifier.  
+
+### Signed Request
+
+The Verifier MAY send a signed request.
+
+The signed request object MAY contain all the parameters listed above except `request`. The signed request object MUST contain an `expected_origins` parameter. 
```

Comment:
This normative statement should not be in an example section.

Reply:
agreed. moved sections around and removed "example flow" section